File: src/modules/user/services/user-delete-avatar.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { StoredFileDeleteService } from 'src/modules/stored-file/services/stored-file-delete.service';
import { UserRequest } from 'src/shared/types/user-request';
import { UserEntity } from '../domain/entities/user.entity';

@Injectable()
export class UserDeleteAvatarService {
  constructor(
    private prisma: PrismaService,
    private storedFileDeleteService: StoredFileDeleteService,
  ) {}
  async execute(id: number, userRequest: UserRequest): Promise<UserEntity> {
    const user = await this.prisma.user.findFirst({
      where: {
        id: id,
        organizationId: userRequest?.organizationId ?? undefined,
        companyId: userRequest?.companyId ?? undefined,
        deletedAt: null,
      },
      include: {
        role: true,
        organization: true,
        company: true,
        avatar: true,
      },
    });

    if (!user) throw new NotFoundException('Arquivo não encontrado!');

    const updatedUser = await this.prisma.user.update({
      data: {
        avatarId: null,
      },
      where: {
        id: user.id,
      },
      include: {
        role: true,
        organization: true,
        company: true,
        avatar: true,
      },
    });

    await this.storedFileDeleteService.execute(user.avatar.uuid);

    return new UserEntity(updatedUser);
  }
}

File: src/modules/user/services/user_delete_avatar_service.py
from django.db.models import Q
from src.shared.types.user_request import UserRequest
from src.modules.user.models import User
from src.shared.exceptions.not_found_exception import NotFoundException
from src.modules.stored_file.services.stored_file_delete_service import (
    StoredFileDeleteService,
)


class UserDeleteAvatarService:
    @classmethod
    def execute(self, id: int, user_request: UserRequest):

        user = User.objects.filter(
            Q(id=id, deleted_at=None)
            & (
                Q(organization_id=user_request.organization_id)
                if user_request and user_request.organization_id
                else Q()
            )
            & (
                Q(company_id=user_request.company_id)
                if user_request and user_request.company_id
                else Q()
            )
        ).first()

        if user is None:
            raise NotFoundException("Usuário não encontrado!")

        if user.avatar is None:
            raise NotFoundException("Arquivo não encontrado!")

        uuid = user.avatar.uuid

        user.avatar = None
        user.save()

        StoredFileDeleteService.execute(uuid)

        return user
